src/tasks/task_list.hpp

```diff
@@ -46,28 +47,102 @@ class TaskList {
   void ClearComplete() {
     auto task = task_list_.begin();
     while (task != task_list_.end()) {
-      if (task->IsComplete()) {
+      if (task->GetStatus() == TaskStatus::complete &&
+          task->GetType() != TaskType::iterative) {
         task = task_list_.erase(task);
       } else {
         ++task;
       }
     }
   }
+  void IterationComplete(const std::string &label) {
+    completed_iters_.insert(label);
+    for (auto &task : task_list_) {
+      if (task.GetLabel() == label) {
+        task.SetStatus(TaskStatus::complete);
+      }
+    }
+  }
+  void ClearIteration(const std::string &label) {
+    auto task = task_list_.begin();
+    while (task != task_list_.end()) {
+      if (task->GetLabel() == label) {
+        task = task_list_.erase(task);
+      } else {
+        ++task;
+      }
+    }
+  }
```

Comment:
I don't know.  I'd like to see some evidence that this is a bottle neck in practice before doing something more complicated.  There are other ways to avoid these sorts of loops too, I think.

Reply:
I agree---don't worry about it for now. It's just something I noticed.